src/util-print.c

```diff
@@ -292,3 +292,10 @@ const char *PrintInet(int af, const void *src, char *dst, socklen_t size)
     }
     return NULL;
 }
+
+void PrintHexString(char *str, size_t size, uint8_t *buf, size_t buf_len)
+{
+    for (size_t x = 0; x < buf_len; x++) {
+        snprintf(str + 2 * x, size - 2 * x, "%02x", buf[x]);
```

Comment:
can we underflow size here?

in general I think its good that we're addressing the bad snprintf patterns, but I feel we're replacing it with another bad one. Can we use a rust func?

Otherwise I think the better pattern is
```
for ( ) {
   char tmp[3];
   snprintf(tmp, sizeof(tmp) ....
   strlcat(str, size, tmp);
}
```
This avoids any buffer size tricks and leaves all validation to the intended funcs